features/support/registration_helpers.rb

```diff
@@ -107,6 +109,49 @@ def complete_operator_name_and_address(registration, address_type)
   complete_address(address_type)
 end
 
+def generate_address_errors(postcode_text)
```

Comment:
We always think small well named methods give a better understanding of whats going on, even if it means the code to do something is now split across multiple methods.

So on this basis I'd suggest going a little further. Currently its not clear to me from the steps that `generate_address_errors()` is actually testing the validation errors on an address lookup page, plus the validation errors on a the manual page. In my opinion (and it is just an opinion so feel free to ignore) I'd tweak it as either

```ruby
# in the steps
test_address_lookup_validations("ROSS KEMP")
# Go to manual address pages:
@world.journey.address_lookup_page.choose_manual_address(
  postcode: "BS1 5AH"
)
test_address_manual_validations
```
or

```ruby
# in the steps
test_address_validations("ROSS KEMP")

# in the helper
def test_address_validations(postcode_text)
  test_address_lookup_validations("ROSS KEMP")
  @world.journey.address_lookup_page.choose_manual_address(
    postcode: "BS1 5AH"
  )
  test_address_manual_validations
end
```

Reply:
Many thanks Alan. I agree this looks more logical. I went with option B because it led to less repetition in the steps. Tests and rubocop still pass. Committed and pushed, will merge now :)